cdm/core/src/main/java/ucar/nc2/dataset/VariableDS.java

```diff
@@ -266,8 +259,17 @@ Array convert(Array data, Set<NetcdfDataset.Enhance> enhancements) {
       if (this.isVariableLength) {
         return data;
       }
-      return scaleMissingUnsignedProxy.convert(data, enhancements.contains(Enhance.ConvertUnsigned),
-          enhancements.contains(Enhance.ApplyScaleOffset), enhancements.contains(Enhance.ConvertMissing));
+      if (enhancements.contains(Enhance.ConvertUnsigned) && unsignedConversion != null) {
+        data = unsignedConversion.convertUnsigned(data);
+      }
+      if (enhancements.contains(Enhance.ApplyScaleOffset) && scaleOffset != null) {
+        data = scaleOffset.removeScaleOffset(data);
+      }
+      if (enhancements.contains(Enhance.ConvertMissing) && convertMissing != null
+          && (dataType == DataType.FLOAT || dataType == DataType.DOUBLE)) {
```

Comment:
maybe you could move the dataType checks `(dataType == DataType.FLOAT || dataType == DataType.DOUBLE)` to `createEnhancements`. That would make it more consistent with what's done for `ScaleOffset`.